src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -271,6 +271,7 @@ protected override void ContextUnloaded(ExecutionContext context)
                     {
                         ExecutionContextState contextState = CurrentContext.GetState<ExecutionContextState>();
                         contextState.NotificationCount += state.NotificationCount;
+                        if (contextState.RequireReturnValue) Push(StackItem.Null);
```

Comment:
This should fix the problem, but please, pay attention to the https://github.com/neo-project/neo/issues/2746. The same thing should also be done for contract that call itself via `System.Contract.Call`.

Reply:
There is no problem here, because in L264 we use a by-reference equality comparison.